checkbox-support/checkbox_support/snap_utils/snapd.py

```diff
@@ -213,6 +213,16 @@ def connect(self, slot_snap, slot_slot, plug_snap, plug_plug):
         if r["type"] == "async" and r["status"] == "Accepted":
             self._poll_change(r["change"])
 
+    def disconnect(self, slot_snap, slot_slot, plug_snap, plug_plug):
+        data = {
+            "action": "disconnect",
+            "slots": [{"snap": slot_snap, "slot": slot_slot}],
+            "plugs": [{"snap": plug_snap, "plug": plug_plug}],
+        }
+        r = self._post(self._interfaces, json.dumps(data))
+        if r["type"] == "async" and r["status"] == "Accepted":
+            self._poll_change(r["change"])
+
```

Comment:
Since this function is exactly the same as `connect`, except its `action` is set to `disconnect` instead of `connect`, I recommend you refactor this.

For example, create a `connect_or_disconnect(self, action, slot_snap, slot_slot, plug_snap, plug_plug)` function that takes `"connect"` or `"disconnect"` as `action`, and does exactly the same as the `connect` function.

Then, you can call `connect_or_disconnect` from `connect` and `disconnect`, turning these functions into one-liners.

You can then revamp a little bit the unit tests (what you wrote is good, just turn this into tests for the new `connect_or_disonnect` function instead, and for the `connect` and `disconnect` functions, just test that they call `connect_or_disonnect` with the proper `action` or something like this)